src/main/java/com/mraof/minestuck/alchemy/CardCaptchas.java

```diff
@@ -0,0 +1,272 @@
+package com.mraof.minestuck.alchemy;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.collect.ImmutableMap;
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.item.MSItems;
+import com.mraof.minestuck.network.MSPacketHandler;
+import com.mraof.minestuck.network.data.ShareCaptchasPacket;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraftforge.event.entity.player.PlayerEvent;
+import net.minecraftforge.event.server.ServerStartedEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.registries.ForgeRegistries;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+
+/**
+ * Handles the creation of, and look up of, captcha codes for each registered item in the given world.
+ * The captcha is arrived at by hashing the registry name string, which gets some additional randomization via world seed.
+ * Two worlds with the same seed will produce the same captcha for a given registered item, expect in some rare instances where there was hash collision.
+ */
+@Mod.EventBusSubscriber(modid = Minestuck.MOD_ID, bus = Mod.EventBusSubscriber.Bus.FORGE)
+public class CardCaptchas
+{
+	//TODO use of the character "#" not canonically accurate and suggests there are more punch holes in a captchalogue card then there actually are
+	
+	public static final String AVAILABLE_CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!?";
+	public static final String EMPTY_CARD_CAPTCHA = "00000000";
+	
+	private static final BiMap<String, String> REGISTRY_MAP = HashBiMap.create();
+	
+	private final Set<String> backupCaptchas = new HashSet<>();
+	private final Set<String> backupCaptchasMaster = new HashSet<>(); //entries are not removed from here
+	
+	private final Random seedRandomizer;
+	
+	public CardCaptchas(ServerLevel level)
+	{
+		this.seedRandomizer = new Random(level.getSeed());
+		generateBackupCaptchas();
+	}
+	
+	@SubscribeEvent
+	public static void serverStarted(ServerStartedEvent event)
+	{
+		CardCaptchas captchas = new CardCaptchas(event.getServer().overworld());
+		captchas.iterateThroughRegistry();
```

Comment:
Perhaps I am overthinking it, but I feel like creating an instance like this is awkward. However I am not sure what a better solution would be

Reply:
One option is to make `iterateThroughRegistry()` static, and create your object for backup captchas inside there instead.

Reply:
Am I overthinking it however?

Reply:
Dunno, but not not by much if so. I'd say I share your feeling on this instance creation. Doing it inside `iterateThroughRegistry()` makes more sense to me since it's first there that it gets used.